server/datastore/mysql/apple_mdm_test.go

```diff
@@ -954,7 +955,20 @@ func configProfileForTest(t *testing.T, name, identifier, uuid string) *fleet.MD
 </dict>
 </plist>
 `, name, identifier, uuid))
-	cp, err := fleet.NewMDMAppleConfigProfile(prof, nil)
+}
+
+func configProfileForTest(t *testing.T, name, identifier, uuid string) *fleet.MDMAppleConfigProfile {
+	prof := configProfileBytesForTest(name, identifier, uuid)
+	cp, err := fleet.NewMDMAppleConfigProfile(configProfileBytesForTest(name, identifier, uuid), nil)
+	sum := md5.Sum(prof) // nolint:gosec // used only to hash for efficient comparisons
+	cp.Checksum = sum[:]
+	require.NoError(t, err)
+	return cp
+}
+
+func teamConfigProfileForTest(t *testing.T, name, identifier, uuid string, teamID uint) *fleet.MDMAppleConfigProfile {
+	prof := configProfileBytesForTest(name, identifier, uuid)
+	cp, err := fleet.NewMDMAppleConfigProfile(configProfileBytesForTest(name, identifier, uuid), &teamID)
 	sum := md5.Sum(prof) // nolint:gosec // used only to hash for efficient comparisons
 	cp.Checksum = sum[:]
 	require.NoError(t, err)
```

Comment:
Same here, this check should be right after `NewMDMAppleConfigProfile` otherwise `cp` could be nil.